internal/data/payments.go

```diff
@@ -640,3 +640,27 @@ func newPaymentQuery(baseQuery string, queryParams *QueryParams, paginated bool,
 	query, params := qb.Build()
 	return sqlExec.Rebind(query), params
 }
+
+// CancelPayments cancels automatically payments that are in "READY" status for an informed time period in days.
+func (p *PaymentModel) CancelPayments(ctx context.Context, sqlExec db.SQLExecuter, periodInDays int64) error {
+	query := `
+		UPDATE payments
+		SET status = 'CANCELED'::payment_status,
+				status_history = array_append(status_history, create_payment_status_history(NOW(), 'CANCELED', NULL))
+		WHERE status = 'READY'::payment_status
+		AND updated_at <= $1
```

Comment:
Please correct if I'm wrong, but I don't believe `updated_at` is a good measure here for cancellation period of a payment. 

Updating any field in a payment can trigger `updated_at` to be changed. 

The only field that seems appropriate is the status history timestamp of the last ready status. e.g. 
`{\"status\": \"READY\", \"timestamp\": \"2023-10-16T21:50:47.911779+00:00\", \"status_message\": null}`